CONTRIBUTING.md

```diff
@@ -0,0 +1,108 @@
+# CONTRIBUTING
+
+Information for anyone who would like to contribute to this repository.
+
+## Repository map
+
+```bash
+├── .git
+├── .github
+│   └── workflows         # Github continuous integration set up
+├── examples              # Python demos: jupyter notebooks
+│   ├── data
+│   ├── example1outputs
+│   ├── example2outputs
+├── glmsingle             # Python implementation
+│   ├── cod
+│   ├── design
+│   ├── gmm
+│   ├── hrf
+│   ├── ols
+│   ├── ssq
+│   └── utils
+├── matlab                # MATLAB implementation
+│   ├── examples          # MATLAB demos
+│   ├── fracridge
+│   └── utilities
+└── tests                 # Python and MATLAB tests
+    └── data
+
+```
+
+## Generic
+
+### Makefile
+
+### pre-commit
+
+## Matlab
+
+### Style guide
+
+### Tests
+
+#### Demos
+
+### Continuous integration
+
+## Python
+
+All the packages required to help with the python development of GLMsingle can
+be installed with:
+
+```bash
+pip install -r requirements_dev.txt
+```
+
+### Style guide
+
+[black](https://black.readthedocs.io/en/stable/) (code formater) and
+[flake8](https://flake8.pycqa.org/en/latest/) (style guide enforcement) are used
+on the test code base.
+
+Ypu can use make to run them automatically with
+
+```bash
+make lint/black # to run black
+make lint/flake8 # to run flake8
+make lint # to run both
+```
```

Comment:
Some of those are for code styling, so it does not have to be done manually. Only applies to the test code base.

Can remove it if you want.